Core/Src/main.c

```diff
@@ -95,6 +99,29 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  // send low to CS pin to start reading from ADC
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  // get data from ADC
+	  uint8_t transmit_data[3] = {0x01, 0x80, 0x00};
+	  uint8_t recieve_data[3];
+	  uint16_t size = 6;
```

Comment:
Why do these values have to be initialized within the while loop... do you know if they would be constantly initialized every iteration of the loop?

If `transmit_data` is not being changed, or SHOULDN'T be changed, consider initializing it as `const`. What const will do is make the compiler complain if this variable is ever modified, it is more a readability and good practice thing bec it tells the reader that this variable isn't meant to change.

Good use of HEX!

So see that you set the size to 6. I think I know why you did that (bec 3 bytes coming in and 3 going out) but in terms of the HAL_SPI_TransmitReceive, it takes number of bytes coming in AND out, so 3. 

Also, maybe size should be const? Remember that the convention for a const value is ALL CAPS.